core/store/src/trie/trie_recording.rs

```diff
@@ -7,22 +7,31 @@ use std::sync::Arc;
 /// A simple struct to capture a state proof as it's being accumulated.
 pub struct TrieRecorder {
     recorded: HashMap<CryptoHash, Arc<[u8]>>,
+    size: usize,
 }
 
 impl TrieRecorder {
     pub fn new() -> Self {
-        Self { recorded: HashMap::new() }
+        Self { recorded: HashMap::new(), size: 0 }
     }
 
     pub fn record(&mut self, hash: &CryptoHash, node: Arc<[u8]>) {
-        self.recorded.insert(*hash, node);
+        match self.recorded.insert(*hash, node.clone()) {
```

Comment:
nit: it is pretty weird to see `clone()` here when node is already owned, in my understanding it is only needed for the `debug_assert_eq!` below which is of very questionable value. I suggest replacing this with `if self.recorded.insert(*hash, node).is_none() { ... }`

Reply:
I removed the debug_assert check... Didn't have it before, not totally worth having now